framework/resource_router.go

```diff
@@ -1,9 +1,110 @@
 package framework
 
-import "context"
+import (
+	"context"
+	"fmt"
 
-func DefaultResourceRouter(rs []Resource) func(ctx context.Context, obj interface{}) ([]Resource, error) {
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/micrologger/loggermeta"
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	"github.com/giantswarm/operatorkit/framework/context/reconciliationcanceledcontext"
+	"github.com/giantswarm/operatorkit/framework/context/resourcecanceledcontext"
+	"github.com/giantswarm/operatorkit/framework/context/updateallowedcontext"
+	"github.com/giantswarm/operatorkit/framework/context/updatenecessarycontext"
+)
+
+type ResourceRouterConfig struct {
+	CustomObjectVersionFunc func(obj interface{}) (string, error)
+	CtxFunc                 func(ctx context.Context, obj interface{}) (context.Context, error)
+	Logger                  micrologger.Logger
+	ResourceFunc            func(ctx context.Context, obj interface{}) ([]Resource, error)
+	VersionBundleVersion    string
```

Comment:
I don't like idea of introducing version bundles concept to the operatorkit.

In general separation of concerns is weird. This struct provides CustomObjectVersion and VersionBundleVersion only to compare it outside.

I'd like to provide something more generic like:

```
// Handles determines if this resource set handles the reconciliation of the object.
Handles func(obj interface{}) bool
```

Reply:
Good point. I like this. Removed the version specific magic and added the `Handles` interface.